File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/annotation/FullName.java
package com.wesleyelliott.kubwa.annotation;

import com.wesleyelliott.kubwa.rule.FullNameRule;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by wesley on 2016/07/28.
 */
@ValidateUsing(FullNameRule.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface FullName {
    int errorMessage();
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/annotation/FullName.kt
package com.wesleyelliott.kubwa.annotation

import com.wesleyelliott.kubwa.rule.FullNameRule

/**
 * Created by wesley on 2016/07/28.
 */
@ValidateUsing(FullNameRule::class)
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.FIELD,
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CONSTRUCTOR,
    AnnotationTarget.VALUE_PARAMETER,
    AnnotationTarget.CLASS,
    AnnotationTarget.FILE
)
annotation class FullName(val errorMessage: Int, val name: String = "fullNameError") {

    @Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
    @Retention(AnnotationRetention.RUNTIME)
    @MustBeDocumented
    annotation class List(vararg val value: FullName)
}
